x/gov/querier.go

```diff
@@ -171,14 +193,41 @@ func queryDeposits(ctx sdk.Context, path []string, req abci.RequestQuery, keeper
 	return bz, nil
 }
 
-// Params for query 'custom/gov/votes'
-type QueryVotesParams struct {
-	ProposalID uint64
+// nolint: unparam
+func queryTally(ctx sdk.Context, path []string, req abci.RequestQuery, keeper Keeper) (res []byte, err sdk.Error) {
+	var params QueryProposalParams
+	err2 := keeper.cdc.UnmarshalJSON(req.Data, &params)
+	if err2 != nil {
+		return nil, sdk.ErrUnknownRequest(sdk.AppendMsgToErr("incorrectly formatted request data", err2.Error()))
+	}
+
+	proposalID := params.ProposalID
+
+	proposal := keeper.GetProposal(ctx, proposalID)
+	if proposal == nil {
+		return nil, ErrUnknownProposal(DefaultCodespace, proposalID)
+	}
+
+	var tallyResult TallyResult
+
+	if proposal.GetStatus() == StatusDepositPeriod {
+		tallyResult = EmptyTallyResult()
+	} else if proposal.GetStatus() == StatusPassed || proposal.GetStatus() == StatusRejected {
+		tallyResult = proposal.GetTallyResult()
+	} else {
+		_, tallyResult = tally(ctx, keeper, proposal)
```

Comment:
This means it's still in the voting phase? Might be helpful to add a short comment here?

Reply:
added